### ISSUE TITLE
Intensity values in print_lines() are mismatched

### ISSUE THREAD

**Issue by sjelabd** [bmcguir2/molsim#57]
The print_ints array that is set for the table output in print_lines() is mismatched with the other print_ arrays that are defined at line ~1480. Perhaps we could make vlsr a required kwarg which would make it easier to use other methods of matching the catalog frequencies and intensities from the simulation?

**Comment by cixue** [comment]
I might have identified two of the potential causes of it.

**(1)** Previously, `l_idxs` and `u_idxs` were calculated from `mol.catalog.frequency`, which was the rest frequency, whereas `sim_l_idxs` and `sim_u_idxs` were calculated from `spectrum.frequency`, which was already shifted by vlsr. However, although in the current version all the `*_idxs` are calculated from the shifted frequency, it leads to inconsistent `ll` and `ul`  in the `print_lines` function compared with that in the `Simulation` object (since `ll` and `ul` in the `Simulation` object are defined in rest frequency and it can results in the  `ll`-`ul` issues as described in (2)). 

Perhaps a better fix for this might be to use `spectrum.freq0` for calculating `sim_l_idxs` and `sim_u_idxs` instead, which is the masked `mol.catalog.frequency`, to keep everything in the rest frequency frame. 

**(2)** If the range defined by the `ll`-`ul` pair provided to the `print_lines` function (no matter they are user-provided values or the default values in the current version) is not totally covered by the range defined by the `sim.ll`-`sim.ul`, lines that are masked (and not simulated) during the construction of the simulation object would be found from the catalog when calling the `print_lines` function. Since the masked lines are not simulated, we won't be able to get the intensity of those lines and therefore a mismatch occurs.

For example, if `mol.catalog.frequency = [1, 2, 3, 4, 5]`, `sim.ll = [2]`, and `sim.ul = [4]`, then `sim.spectrum.freq0 = [2, 3, 4]`. But if a user provides a `ll`-`ul` pair of `1` and `3` to the `print_lines` function, it will find line properties of the lines with frequencies `[1, 2, 3]` from the catalog but the line intensities of the lines with frequencies `[2, 3]` from the simulated spectrum.

A suggested fix is to mask the catalog with `sim.ll` and `sim.ul` before masking with the `ll` and `ul` arguments in the `print_lines` function.

If @bmcguir2 and @laserkelvin are okay with this fix, I can submit a pull request for this.

**Comment by laserkelvin** [comment]
I think @sjelabd had started working on a solution already, but I'm not sure what the state of that is?

If not, I agree with masking the catalog with `sim.ll` and `sim.ul`, although I'm not sure if that will require a large change in the code. If it does, we should probably put some unit tests in place before making such a change.

**Comment by sjelabd** [comment]
Yes, I agree with point 1) from @cixue. My solution was to shift the catalog frequencies when creating the mask for every Simulation object, which appears to solve the issue on my end. Haven't pushed the update since I've also made another change to how print_lines() worked and @bmcguir2 wanted to make a small change to that.

**Comment by cixue** [comment]
I just took a look at @sjelabd's pull request #61  and I think it does fix the first issue that I pointed out. However, we still need to mask out the catalog in advance though (i.e. point (2)) and it shouldn't take more than a few lines.

In addition, I just found another bug that `uls` and `lls` have not been properly set up if `ul` and `ll` are specified by user when calling the `print_lines` function. I can submit a pull request after #61 is merged.

**Comment by laserkelvin** [comment]
Merged #61, so if you want to make a PR for point 2 @cixue please do so after you pull the latest development code
